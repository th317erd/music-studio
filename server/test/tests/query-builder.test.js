const { QueryBuilder } = require('../../base/query-builder')({ test: true });

describe('QueryBuilder', function() {
  it('do a join with a query builder', function() {
    var fullQuery = new QueryBuilder();
    fullQuery = fullQuery.model('fluid').join((query) => {
      return query.model('fluid_sources').fluid_sources_fluid_id.matches('fluid.id');
    }).and((query) => {
      return query.model('fluid_sources').heartbeats_heartbeat_id.eq('HB:b9c62f26-7e1b-44fc-89da-b39400b38546');
    }).order('fluid_sources.fluid_sources_fluid_order DESC').finalize();

    var sqlString = fullQuery.toSQL();
    expect(sqlString).toMatchSnapshot();
  });

  it('serialize a query builder', function() {
    var query = new QueryBuilder().model('heartbeat').id.eq('test').name.like('*derp*').time.gt('2018-09-21').order('id').limit(10).offset(1).finalize();
    var serialized = query.serialize();
    expect(serialized).toMatchSnapshot();
  });

  it('deserialize a query builder', function() {
    var serialized = '["id ASC",10,1,100]heartbeat:id="test"&heartbeat:name%"*derp*"&heartbeat:time>"2018-09-21"';
    var query = QueryBuilder.unserialize(serialized);
    expect(query.serialize()).toBe(serialized);
  });
});
