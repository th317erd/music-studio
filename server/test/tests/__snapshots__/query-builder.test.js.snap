// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryBuilder do a join with a query builder 1`] = `"SELECT * FROM \\"fluid\\" JOIN \\"fluid_sources\\" ON (\\"fluid_sources\\".\\"fluid_sources_fluid_id\\"=\\"fluid\\".\\"id\\") WHERE \\"fluid_sources\\".\\"heartbeats_heartbeat_id\\"='HB:b9c62f26-7e1b-44fc-89da-b39400b38546' ORDER BY \\"fluid_sources\\".\\"fluid_sources_fluid_order\\" DESC"`;

exports[`QueryBuilder serialize a query builder 1`] = `"[\\"id ASC\\",10,1,100]heartbeat:id=\\"test\\"&heartbeat:name%\\"*derp*\\"&heartbeat:time>\\"2018-09-21\\""`;
